src/main/java/org/eclipsefoundation/adopters/service/impl/DefaultAdopterService.java

```diff
@@ -58,123 +56,110 @@
-				while (this.running) {
-					WatchKey key = null;
-					try {
-						key = watcher.take();
-						// if there was an error while watching for changes, try again
-						if (key == null) {
-							continue;
-						}
-
-						// pass a filtered list of events to be processed
-						@SuppressWarnings("unchecked")
-						List<WatchEvent<Path>> events = key.pollEvents().stream()
-								.filter(e -> e.kind() != OVERFLOW && e.kind().type() == Path.class)
-								.map(e -> (WatchEvent<Path>) e)
-								.filter(e -> ((Path) e.context()).getFileName().toString().equals(adoptersName))
-								.collect(Collectors.toList());
-						processEvents(events);
-					} catch (InterruptedException e) {
-						// represents shutdown state, so end processing
-						Thread.currentThread().interrupt();
-					} finally {
-						// return the key to the watch service
-						if (key != null && key.isValid()) {
-							key.reset();
-						}
-					}
+				key = watcher.take();
+				// if there was an error while watching for changes, try again
+				if (key == null) {
+					continue;
+				}
+
+				// pass a filtered list of events to be processed
+				key.pollEvents().stream().filter(e -> e.kind() != OVERFLOW && e.kind().type() == Path.class)
+						.map(e -> (WatchEvent<Path>) e)
+						.filter(e -> ((Path) e.context()).getFileName().equals(adoptersLocation.getFileName()))
+						.forEach(this::processEvent);
+			} catch (@SuppressWarnings("java:S2142") InterruptedException e) {
+				LOGGER.debug("Ending watch services, as thread was interrupted and closing");
+				try {
+					this.watcher.close();
+				} catch (IOException e1) {
+					LOGGER.error("Error while closing the watch service", e1);
 				}
+				break;
 			} finally {
-				// release the lock once leaving the current block
-				this.lock.unlock();
+				// return the key to the watch service
+				if (key != null && key.isValid()) {
+					key.reset();
+				}
 			}
 		}
+
 	}
 
-	private void processEvents(List<WatchEvent<Path>> events) {
-		// process each of the queued events
-		for (WatchEvent<Path> event : events) {
-			// clear the adopter list if the event is a deletion event
-			if (event.kind() == ENTRY_DELETE) {
-				LOGGER.debug("Detected the deletion of adopters file, emptying array");
-				synchronized (this) {
-					this.adopters = new ArrayList<>();
-				}
-			} else {
-				// indicates an update or created file on the given
-				readInFile(Paths.get(adoptersLocation, event.context().getFileName().toString()));
+	private void processEvent(WatchEvent<Path> event) {
+		// clear the adopter list if the event is a deletion event
+		if (event.kind() == ENTRY_DELETE) {
+			LOGGER.debug("Detected the deletion of adopters file, emptying array");
+			synchronized (this) {
+				this.adopters = new ArrayList<>();
 			}
+		} else {
+			// indicates an update or created file on the given
+			readInFile(adoptersLocation);
 		}
+
 	}
 
 	private void readInFile(Path adoptersPath) {
```

Comment:
Do not mix IO and class state change. Make this method static and return a List<Adopter>. It will make it more readable and the caller code will be clearer as well.